### DISCUSSION TITLE
DR plans for IRIS Production Server

### DISCUSSION THREAD

**Discussion by TPower2112** [dfir-iris/iris-web#321] [General]
Hi,

We are planning for Disaster Recovery.  For an IRIS migration to a new server, would it just involve saving and copying all the container images and any persistent volumes to a new instance?

Something similar to this:
https://www.linuxandubuntu.com/home/migrate-docker-containers-to-new-server

Has anyone tried this?
Thanks for your time and attention.

TP

**Comment by m-terlinde** [comment]
Hey hey,

I have two things in my head right now:

- you should be able to just copy the database volume and spin up new images/containers
- the [documentation](https://docs.dfir-iris.org/operations/upgrades/)  shows a way to export the database into an archive. With that you also should be able to initialise a fresh container setup

I don't think copying the containers itself is the best way and contradicts the idea of containers.

I would build IRIS on the DR platform, but keep it uninitialised.

Cheers,
Matthias

**Comment by TPower2112** [comment]
I migrated the database and app volumes along with the docker images and ssl certs. 

That worked for me.

**Comment by TPower2112** [comment]
Here are the steps for restoring the db container on a DR server which was easier than copying the app and db volumes.

#back up the production db container
docker exec iriswebapp_db pg_dump -U postgres iris_db | \gzip > ../iris_db_backup.gz

#install docker, docker compose, and IRIS on DR server

#copy the backup to the DR Server in the iris-web directory

#stop containers on the DR server
docker compose stop

#remove db container volumes
docker compose down db --volumes

#spin up db container
docker compose up db

#obtain docker container ID for db
docker container list

#restore the db data
zcat ../iris_db_backup.gz | docker exec -i containerID psql -U postgres -d iris_db 

#spin up the rest of the containers w force-recreate flag to remove network ID
#https://stackoverflow.com/questions/53347951/docker-network-not-found

docker compose up --force-recreate

**Reply by L88p** [reply]
Just adding that the backup of the db retain the custom modules configurations, but the modules themselves must be reinstalled manually (and if necessary also backed up separatedly).

https://docs.dfir-iris.org/development/modules/quick_start/processor/#installing-and-registering-the-module
